docs/rules/no-restricted-paths.md

```diff
@@ -0,0 +1,38 @@
+# no-restricted-paths - Restrict which files can be imported in a given folder
+
+Some projects contain files which are not always meant to be executed in the same environment.
+For example consider a web application that contains specific code for the server and some specific code for the browser/client. In this case you don’t want to import server-only files in your client code.
+
+In order to prevent such scenarios this rule allows you to define restricted zones where you can forbid files from imported if they match a specific path.
+
+## Rule Details
+
+This rule has tow options. The first option is an array to define all restricted zones. The second option is optional and allows you to override the `basePath` which is the current working directory by default. All paths defined in the first option are either absolute paths or relative to the `basePath`.
+
+
+### Examples
+
+Given the following folder structure:
+
+```
+my-project
+├── client
+│   └── foo.js
+│   └── baz.js
+└── server
+    └── bar.js
+```
+
+and the current file being linted is `my-project/client/foo.js`.
+
+The following patterns are considered problems when configuration set to `[ { "target": "./client", "from": "./server" } ]`:
+
+```js
+import bar from '../server/bar';
+```
+
+The following patterns are not considered problems when configuration set to `[ { "target": "./client", "from": "./server" } ]`:
+
+```js
+import foo from '../server/baz';
```

Comment:
I'm not sure I see how this is different, except for the fact that `server/baz` does not exist.